x/incentives/keeper/distribute_test.go

```diff
@@ -587,3 +587,105 @@ func (suite *KeeperTestSuite) TestGetValidConcentratedLiquidityGauge() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestDistributeConcentratedLiquidityInternal() {
+	defaultGauge := perpGaugeDesc{
+		lockDenom:    defaultLPDenom,
+		lockDuration: defaultLockDuration,
+		rewardAmount: sdk.Coins{sdk.NewInt64Coin(defaultRewardDenom, 3000)},
+	}
+	doubleLengthGauge := perpGaugeDesc{
+		lockDenom:    defaultLPDenom,
+		lockDuration: 2 * defaultLockDuration,
+		rewardAmount: sdk.Coins{sdk.NewInt64Coin(defaultRewardDenom, 3000)},
+	}
+	defaultGaugeStartTime := suite.Ctx.BlockTime()
+
+	type distributeConcentratedLiquidityInternalTestCase struct {
+		name            string
+		poolId          uint64
+		sender          sdk.AccAddress
+		incentiveDenom  string
+		incentiveAmount sdk.Int
+		emissionRate    sdk.Dec
+		startTime       time.Time
+		minUptime       time.Duration
+		expectedCoins   sdk.Coins
+		gauges          []perpGaugeDesc
+		expectedError   bool
+	}
+
+	testCases := []distributeConcentratedLiquidityInternalTestCase{
+		{
+			name:            "Valid Case: valid incentive Record with valid Gauge",
+			poolId:          1,
+			sender:          suite.TestAccs[0],
+			incentiveDenom:  defaultRewardDenom,
+			incentiveAmount: sdk.NewInt(100),
+			emissionRate:    sdk.NewDec(1),
+			startTime:       defaultGaugeStartTime.Add(time.Hour),
+			minUptime:       time.Hour * 24,
+			gauges:          []perpGaugeDesc{defaultGauge, doubleLengthGauge},
+
+			expectedCoins: sdk.NewCoins(sdk.NewCoin(defaultRewardDenom, sdk.NewInt(100))),
+		},
+		{
+			name:            "Invalid Case: valid incentive Record with invalid Gauge",
+			poolId:          1,
+			sender:          suite.TestAccs[0],
+			incentiveDenom:  defaultRewardDenom,
+			incentiveAmount: sdk.NewInt(200),
+			emissionRate:    sdk.NewDec(2),
+			startTime:       defaultGaugeStartTime,
+			minUptime:       time.Hour * 24,
+			gauges:          []perpGaugeDesc{},
+
+			expectedError: true,
+		},
+		{
+			name:            "Invalid Case: invalid incentive Record with valid Gauge",
+			poolId:          1,
+			sender:          suite.TestAccs[0],
+			incentiveDenom:  defaultRewardDenom,
+			incentiveAmount: sdk.NewInt(200),
+			emissionRate:    sdk.NewDec(2),
+			startTime:       defaultGaugeStartTime,
+			minUptime:       time.Hour * 24,
+			gauges:          []perpGaugeDesc{defaultGauge, doubleLengthGauge},
+
+			expectedError: true,
+		},
```

Comment:
I might be missing it, but what makes this an invalid incentive record

Reply:
it has to do with the starttime, here the startime is before blockTime. I'll make it a little more verbose